File: algorithms_2/queue_2_stacks.py
def enqueue(s1, s2, item):
  s2.append(item)

def dequeue(s1, s2):
  if len(s2) == 0:
    return
  s1 = s2.copy()
  s1.reverse()
  s1.pop()
  s1.reverse()
  for i in range(len(s1)):
    s2[i] = s1[i]
  s2.pop()

st1 = []
st2 = []
enqueue(st1, st2, 3)
print(st2)
enqueue(st1, st2, 76)
print(st2)
enqueue(st1, st2, 35)
print(st2)
enqueue(st1, st2, 71)
print(st2)
enqueue(st1, st2, 1)
print(st2)

dequeue(st1, st2)
print(st2)
dequeue(st1, st2)
print(st2)
dequeue(st1, st2)
print(st2)
dequeue(st1, st2)
print(st2)
dequeue(st1, st2)
print(st2)
dequeue(st1, st2)
print(st2)

File: cplusplus/oop/stack/queue_2_stacks.cpp
#include "queue_2_stacks.h"

q_2_s::q_2_s(int capacity): capacity(capacity), s1(capacity), s2(capacity) {
}

void q_2_s::enqueue(int item) {
  s1.push(item);
}

void q_2_s::dequeue() {
  int s1_size = s1.get_size();
  for (int i = 0; i < s1_size; i++) {
    int popped_item = s1.pop();
    s2.push(popped_item);
  }
  s2.pop();

  int s2_size = s2.get_size();
  for (int i = 0; i < s2_size; i++) {
    int popped_item = s2.pop();
    s1.push(popped_item);
  }
}

void q_2_s::print() const {
  s1.print();
}
